annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastErrorReporter.java

```diff
@@ -62,38 +61,44 @@ private void reportErrorMessage(String message, Stream<Element> elements) {
         elements.forEach(element -> messager.printMessage(Diagnostic.Kind.ERROR, message, element));
     }
 
-    void reportDuplicatePackages(Stream<ArrayList<FastPackageElement>> fastPackages) {
-        fastPackages.forEach(fastPackageElements -> {
-            String packageName = fastPackageElements.iterator().next().getPackageName();
-            boolean containsAnonymousPackage = false;
-            for (FastPackageElement packageElement : fastPackageElements) {
-                if (packageElement instanceof AnonymousFastPackage) {
-                    containsAnonymousPackage = true;
-                    break;
-                }
-            }
-            if (!containsAnonymousPackage) {
-                reportErrorMessage(
-                    String.format(DUPLICATE_PACKAGE_TEMPLATE, packageName),
-                    fastPackageElements.stream()
-                        .map(fastPackage -> ((NamedFastPackage) fastPackage).getPackageElement()));
-            } else {
-                reportErrorMessage(
-                    String.format(ANONYMOUS_EQUALS_NAMED_TEMPLATE, packageName),
-                    fastPackageElements.stream().filter(packageElement -> packageElement instanceof NamedFastPackage)
-                        .map(namedPackage -> ((NamedFastPackage) namedPackage).getPackageElement())
-                );
+    @Override
+    public void reportInvalidPackageName(FastPackageElement fastPackage) {
+        reportErrorMessage(
+            String.format(INVALID_PACKAGE_TEMPLATE, fastPackage.getPackageName()),
+            Stream.of(((NamedFastPackage) fastPackage).getPackageElement())
+        );
+    }
+
+    @Override
+    public void reportDuplicatePackages(Collection<FastPackageElement> fastPackages) {
+        boolean containsAnonymousPackage = false;
+        String fastPackageName = fastPackages.iterator().next().getPackageName();
+        for (FastPackageElement fastPackage : fastPackages) {
+            if (fastPackage instanceof AnonymousFastPackage) {
+                containsAnonymousPackage = true;
+                break;
             }
-        });
+        }
+        if (containsAnonymousPackage) {
```

Comment:
The logic should be different: we should print different message for anonymous package and for named package, but not different message if have anonymous package